packages/stex-react-renderer/src/lib/PerSectionQuiz.tsx

```diff
@@ -15,6 +22,87 @@ export function handleViewSource(problemUri: string) {
     if (sourceLink) window.open(sourceLink, '_blank');
   });
 }
+function getSections(tocElems: FTML.TOCElem[]): string[] {
+  const sectionIds: string[] = [];
+  for (const tocElem of tocElems) {
+    if (tocElem.type === 'Section') {
+      sectionIds.push(tocElem.uri);
+    }
+    if ('children' in tocElem) {
+      sectionIds.push(...getSections(tocElem.children));
+    }
+  }
+  return sectionIds;
+}
+export async function getAllConceptUrisForCourse(
```

Comment:
this needs to be done in the backend (with caching). doing this on the frontend would be slow and expensive